src/charm.py

```diff
@@ -83,6 +99,19 @@ def __init__(self, *args):
             self._on_loki_push_api_endpoint_departed,
         )
 
+    def update_alerts_rules(self, _):
+        """Copy alert rules from relations and save them to disk."""
+        rules = self._scrape.alerts()
+        shutil.rmtree(self._metrics_rules_dest_path)
+        shutil.copytree(self._metrics_rules_src_path, self._metrics_rules_dest_path)
+        for topology_identifier, rule in rules.items():
+            file_handle = pathlib.Path(
+                self._metrics_rules_dest_path, "juju_{}.rules".format(topology_identifier)
+            )
+            file_handle.write_text(yaml.dump(rule))
```

Comment:
If we get another push, this wouldn't hurt to change. The name is ok, but it makes me feel like it's a handle which should be closed so it doesn't leak rather than `Path` hiding it.